gravity-forms/gw-require-unique-values.php

```diff
@@ -85,6 +89,34 @@ public function validate( $result, $value, $form, $field ) {
 		return $result;
 	}
 
+	public function validate_list( $result, $value ) {
+		$rows      = count( $value );
+		$is_unique = true;
+
+		// Multi-Column List
+		if ( is_array( $value[0] ) ) {
+			foreach ( array_keys( $value[0] ) as $column_name ) {
+				$column_values = array_column( $value, $column_name );
+				// Count unique values in each column.
+				if ( $rows != count( array_unique( $column_values ) ) ) {
+					$is_unique = false;
+				}
+			}
+		} else {
+			// Count unique values in the Single Column List.
+			if ( $rows != count( array_unique( $value ) ) ) {
+				$is_unique = false;
+			}
+		}
+
+		if ( $result['is_valid'] && ! $is_unique ) {
+			$result['is_valid'] = false;
+			$result['message']  = $this->_args['validation_message'] . ' All columns should have unique values.';
```

Comment:
Like the idea of providing additional clarification about the validation error; however, the user already has the ability to define their own validation message and this would prevent them from customizing that in its entirety. For simplicity, I would simply use the default validation message that the user can modify.

Reply:
Got it. Updated